docs/development/releasing.rst

```diff
@@ -630,6 +630,9 @@ applies both for regular release *and* release candidates are the same
       $ git checkout <maintenance branch name>
       $ git merge --no-ff tmp-release-v<version>
       $ git tag -s "v<version>" -m "Tagging v<version>"
+      $ git clean -dfx
+      $ umask 0022
```

Comment:
I think it's worth adding just a sentence or two in the text explaining why you're doing this - it's not really obvious from this what the point is.